### PR TITLE
expression: implement vectorized evaluation for `builtinCRC32Sig`

### PR DESCRIPTION
**Pull request by TommyCpp**
### What problem does this PR solve? 
implement vectorized evaluation for builtinIsIPv4MappedSig , for #12103 

### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinMathFunc -run=BenchmarkVectorizedBuiltinMathFunc -args "builtinCRC32Sig"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinMathFunc/builtinCRC32Sig-VecBuiltinFunc-12               47541             26448 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinCRC32Sig-NonVecBuiltinFunc-12            15898             81080 ns/op           16736 B/op        804 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      3.581s
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_math_vec.go [hunk 1]
```diff
@@ -658,11 +659,30 @@ func (b *builtinRoundWithFracIntSig) vectorized() bool {
...
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := range i64s {
+		if buf.IsNull(i) {
+			i64s[i] = 0
+			result.SetNull(i, true)
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]: We can use result.MergeNull to speed up NullBitMap operator.

File: expression/builtin_math_vec.go [hunk 2]
```diff
@@ -658,11 +659,30 @@ func (b *builtinRoundWithFracIntSig) vectorized() bool {
...
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := range i64s {
+		if buf.IsNull(i) {
+			i64s[i] = 0
```
Comment (wshwsh12) [inline expression/builtin_math_vec.go]: Don't need to set `i64s[i]=0`. If buf.IsNull(i) is true, the value is meaningless.

File: expression/builtin_math_vec.go (hunk 1 above)
Comment (TommyCpp) [reply expression/builtin_math_vec.go]: Thanks for the hint! I will change accordingly.

File: expression/builtin_math_vec.go (hunk 2 above)
Comment (TommyCpp) [reply expression/builtin_math_vec.go]: And I guess the `ResizeInt64` will initial all the value to 0? so this line is indeed redundant. I will remove it.

**Pull request merged**